benchmark/src/main/scala/keycloak/scenario/KeycloakScenarioBuilder.scala

```diff
@@ -285,19 +285,20 @@ class KeycloakScenarioBuilder {
   }
 
   private def getServiceAccountTokenExec(): ChainBuilder = {
-    exec(http("Get service account token")
+    // Store the current timestamp BEFORE it's actually sent (as we don't know how long the request will take or if it will succeed at all)
+    exec(_.set("requestEpoch", System.currentTimeMillis()))
+    .exec(http("Get service account token")
       .post(TOKEN_ENDPOINT)
       .formParam("grant_type", "client_credentials")
       .formParam("client_id", "gatling")
       .formParam("client_secret", "${clientSecret}")
       .check(
         jsonPath("$..access_token").find.saveAs("token"),
-        jsonPath("$..expires_in").find.saveAs("expiresIn"),
-        header("Date").saveAs("tokenTime")
+        jsonPath("$..expires_in").find.saveAs("expiresIn")
       ))
       .exitHereIfFailed
       .exec(s => {
-        s.set("accessTokenRefreshTime", ZonedDateTime.parse(s("tokenTime").as[String], DATE_FMT).toEpochSecond * 1000)
+        s.set("accessTokenRefreshTime", "${requestEpoch}")
```

Comment:
I like the idea to capture the time before sending the request as an attribute.

I ran the Scenario against a Quarkus and Wildfly distribution (SNAPSHOT) of Keycloak. The login works, but then an exception "j.l.NumberFormatException: For input string: "${requestEpoch}"" appears in the log.

A assume this is due that s.set() doesn't replace the attribute as the code change above assumes. 

What works instead in my local tests is fetching the attribute without the replacement. Please give that one a try and update this PR. 

```suggestion
        s.set("accessTokenRefreshTime", s.attributes.get("requestEpoch"))
```

Please apply this also in the other similar places.

Reply:
All done, rebased and pushed.
Thanks for catching the error (obviously, pushing a `String` where a `long` is expected does not play nice - weird that I did not see the error in the first place, though).

Reply:
Thank you! You can now mark this as resolved.